rosdep/base.yaml

```diff
@@ -1411,6 +1411,13 @@ gnuplot:
 gnuplot-x11:
   debian: [gnuplot-x11]
   ubuntu: [gnuplot-x11]
+gnutls:
+  alpine: [gnutls-dev]
+  debian: [libgnutls28-dev]
+  fedora: [gnutls-devel]
+  opensuse: [libgnutls-devel]
+  rhel: [gnutls-devel]
+  ubuntu: [libgnutls28-dev]
```

Comment:
> I'm surprised to find it is not currently documented but the current convention for rosdep keys is to name them after the Debian/Ubuntu package name.

There is no common pattern in the key names. `boost` and `eigen` are also refering to the `lib*` versions. But in this case there is no stand-alone program with a similar name

> I think we should adhere to it here in order to disambiguate with the gnutls-bin package.

Can I name it `libgnutls`? :-) I think the `-dev` suffix is redundant since rosdep does not differentiate runtime and devel packages. I would not encode the version (28) in the package name since this will change with time.

Reply:
> There is no common pattern in the key names. boost and eigen are also refering to the lib* versions. But in this case there is no stand-alone program with a similar name

The convention is to use the ubuntu package name by default. When we change our conventions we grandfather in existing rules such that we don't cause disruption to existing users. That's why you'll see older formats. 

> Can I name it libgnutls? :-) I think the -dev suffix is redundant since rosdep does not differentiate runtime and devel packages. 

Please follow the convention. Rosdep does not know or care what's in the packages. However users do care. If you are looking for a resources that has the header files the `-dev` suffix is critical to know that it's available. If you make a `libgnutls`  key it would be expected to refer to the [runtime libraries](https://packages.debian.org/bullseye/libgnutlsxx28) and not the development headers. 

> I would not encode the version (28) in the package name since this will change with time.

Our general policy is that if it's important enough for debian/ubuntu to encode the version then it's likely important enough for us to encode the version. If you can make the case and demonstrate that the future packaging plans for this key will be forwards compatible we can consider. However looking forward right now. All [future debian packaging targets](https://packages.debian.org/bullseye/libgnutls28-dev) seems to include the 28 version suggesting that this key will be valid for many years to come.